### PR TITLE
[IMP] web: update owl to last version

### PR DESCRIPTION
**Pull request by ged-odoo**
This owl update is significant: it introduces a change in the way
components are rendered: now, child components check if there is a
current rendering somewhere in their ancestors. If this is the case, the
rendering will be delayed.  This prevents lots of issues where a
child components may be in some kind of corrupted state after some state
changes.

Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Review by sdegueldre — APPROVED** [review]

File: addons/web/static/src/views/helpers/model.js [hunk 1]
```diff
@@ -98,10 +98,14 @@ export function useModel(ModelClass, params, options = {}) {
     const orm = model.orm;
     let sampleORM = globalState.sampleORM;
     const user = useService("user");
+    let started = false;
     async function load(props) {
         model.orm = orm;
         const searchParams = getSearchParams(props);
         await model.load(searchParams);
+        if (started) {
+            model.notify();
```
Comment (aab-odoo) [inline addons/web/static/src/views/helpers/model.js:107]: shouldn't this be done line 117 instead?

File: addons/web/static/src/views/pivot/pivot_model.js [hunk 2]
```diff
@@ -1521,9 +1521,11 @@ export class PivotModel extends Model {
...
             metaData.fields[fieldName].type === "boolean"
         ) {
-            return value === undefined ?
-                this.env._t('None')
-                : (value ? this.env._t('Yes') : this.env._t('No'));
+            return value === undefined
+                ? this.env._t("None")
+                : value
+                ? this.env._t("Yes")
+                : this.env._t("No");
```
Comment (aab-odoo) [inline addons/web/static/src/views/pivot/pivot_model.js:1528]: remove this diff

File: addons/web/static/tests/search/search_panel_tests.js [hunk 3]
```diff
@@ -990,9 +990,9 @@ QUnit.module("Search", (hooks) => {
         const asustekPromise = promise;
         await click(getCategory(target, 1));
 
-        // 'asustek' should be selected, but there should still be 3 records
+        // 'asustek' should not be selected yet, but there should still be 3 records
```
Comment (aab-odoo) [inline addons/web/static/tests/search/search_panel_tests.js:993]: **and** there should still be...

File: addons/web/static/tests/search/search_panel_tests.js [hunk 4]
```diff
@@ -1001,9 +1001,9 @@ QUnit.module("Search", (hooks) => {
         const agrolaitPromise = promise;
         await click(getCategory(target, 2));
 
-        // 'agrolait' should be selected, but there should still be 3 records
+        // 'agrolait' should not be selected yet, but there should still be 3 records
```
Comment (aab-odoo) [inline addons/web/static/tests/search/search_panel_tests.js:1004]: same

**Review by aab-odoo — APPROVED** [review]
I don't know about the diff in owl2 compatibility. If you want me to really review it, I would need more time/explanations.

**Comment by sdegueldre** [comment]
@robodoo r+

**Pull request closed without merging**